### PR TITLE
Repository API design and implementation

### PR DESCRIPTION
**Pull request by markpatton**

### CODE REVIEW

File: pass-core-main/src/test/java/org/eclipse/pass/object/IntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019, Yahoo
```
Comment (jabrah) [inline pass-core-main/src/test/java/org/eclipse/pass/object/IntegrationTest.java:2]: Should this license statement be changed to our project default statement?

**Review by jabrah — APPROVED** [review]
Not too much from my perspective outside of the open question about the filter implementation and some minor comment editing. I'm amused by some of the seemingly gory Elide detail, like the to_json_api_doc method and tx handling

I do prefer the idea of the client impl triggering hooks and security checks as expected, but I'm not entirely sure if those come any usage caveats

File: pass-core-object-service/src/main/java/org/eclipse/pass/object/RSQL.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    public static String equals(String name, String value) {
+        return comparison(name, "==", value);
+    }
+
+    public static String notEquals(String name, String value) {
+        return comparison(name, "!=", value);
+    }
+
+    public static String in(String name, String... values) {
```
[the remark this answers is not in the document]
Comment (markpatton) [reply pass-core-object-service/src/main/java/org/eclipse/pass/object/RSQL.java:40]: Good question. My thought was to wait and then respond to user needs. There are also a bunch of questions about full text search to think about. Using glob patterns is probably a bad idea.

File: pass-core-main/src/test/java/org/eclipse/pass/object/IntegrationTest.java (hunk 1 above)
Comment (markpatton) [reply pass-core-main/src/test/java/org/eclipse/pass/object/IntegrationTest.java:2]: Right. Missed that.

**Pull request merged**